server/service/src/document/update_trigger.rs

```diff
@@ -0,0 +1,67 @@
+use std::str::FromStr;
+
+use chrono::NaiveDate;
+use repository::{StorageConnection, Document, NameRowRepository, NameRow, NameType, Gender as NameRowGender};
+
+extern crate serde;
+extern crate schemafy_core;
+extern crate serde_json;
+
+use serde::{Serialize, Deserialize};
+
+use super::document_service::DocumentInsertError;
+
+
+schemafy::schemafy!(
+  "src/document/schemas/patient.json"
+);
+
+/// Callback called when the document has been updated
+pub fn document_updated(con: &StorageConnection, store_id: &str, doc: &Document) -> Result<(), DocumentInsertError> {
+  match doc.r#type.as_str() {
+    "Patient" => {
+      let patient: Patient = serde_json::from_value(doc.data.clone()).map_err(|err| DocumentInsertError::InternalError(format!("Invalid date of birth format: {}", err)))?;
+      let contact = patient.contact_details.get(0);
+      let date_of_birth =  match patient.date_of_birth{
+        Some(date_of_birth ) => Some(NaiveDate::from_str(&date_of_birth).map_err(|err| DocumentInsertError::InternalError(format!("Invalid date of birth format: {}", err)))?),
+        None => None
+      };
+      let address = patient.addresses.get(0);
+      let name_repo = NameRowRepository::new(con);
+      name_repo.upsert_one(&NameRow {
+        id: format!("{}@{}", patient.id, store_id),
```

Comment:
I wonder if this will be annoying - when trying to fetch a patient by id. is it not enough to have the patient `id` and the `supplying_store_id`?

Reply:
I changed it [9aa4d3f](https://github.com/openmsupply/open-msupply/pull/46/commits/9aa4d3ffc3ea8ec334ee3212886eaddd2ce7a8b9)